c++11/src/grpc_transporter.cpp

```diff
@@ -13,34 +19,64 @@ static std::string HostPortOf(const LightStepTracerOptions& options) {
 }
 
 //------------------------------------------------------------------------------
-// Constructor
+// GrpcTransporter
 //------------------------------------------------------------------------------
-GrpcTransporter::GrpcTransporter(spdlog::logger& logger,
-                                 const LightStepTracerOptions& options)
-    : logger_{logger},
-      client_{grpc::CreateChannel(
-          HostPortOf(options),
-          (options.collector_encryption == "tls")
-              ? grpc::SslCredentials(grpc::SslCredentialsOptions())
-              : grpc::InsecureChannelCredentials())},
-      report_timeout_{options.report_timeout} {}
+namespace {
+// GrpcTransporter sends ReportRequests to the specified host via gRPC.
+class GrpcTransporter : public SyncTransporter {
+ public:
+  GrpcTransporter(spdlog::logger& logger, const LightStepTracerOptions& options)
+      : logger_{logger},
+        client_{grpc::CreateChannel(
+            HostPortOf(options),
+            (options.collector_encryption == "tls")
+                ? grpc::SslCredentials(grpc::SslCredentialsOptions())
+                : grpc::InsecureChannelCredentials())},
+        report_timeout_{options.report_timeout} {}
+
+  opentracing::expected<void> Send(
+      const google::protobuf::Message& request,
+      google::protobuf::Message& response) override {
+    grpc::ClientContext context;
+    collector::ReportResponse resp;
+    context.set_fail_fast(true);
+    context.set_deadline(std::chrono::system_clock::now() + report_timeout_);
+    auto status = client_.Report(
+        &context, dynamic_cast<const collector::ReportRequest&>(request),
+        dynamic_cast<collector::ReportResponse*>(&response));
+    if (!status.ok()) {
```

Comment:
the `status` object has an associated code number. Is there a way to use that?

Reply:
I could probably use it to map to better standard [error codes](http://www.cplusplus.com/reference/system_error/errc/). Using them directly would mean having to set up a gRPC specific [error category](http://en.cppreference.com/w/cpp/error/error_category).